bin/check_autogenerated_files.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# ###
+# check_autogenerated_files.sh
+# ###
+# Makes sure that autogenerated files were regenerated after making changes to the source.
+# ###
+echo "Downloading configlet"
+bin/fetch-configlet
+echo "Running 'configlet generate'"
+bin/configlet generate
+
+echo "Checking git diff"
+git diff --exit-code HEAD
+GIT_DIFF_EXIT_CODE="$?"
+
+if [ "$GIT_DIFF_EXIT_CODE" -ne 0 ]
+then
+  echo "Autogenerated files check failed. Please run \`configlet generate\` and commit in the changes."
+  exit 1;
```

Comment:
If I set `set -euo pipefail` in this script, then it always exists after the git diff before echoing the explanation :( so I didn't add it. We risk it "passing" because something else goes wrong.

Reply:
You could also grab the output, which should always succeed and exist depending on the content

```bash
GIT_DIFF=$(git diff)

if [ -z "$GIT_DIFF" ]
then 
  echo "Autogenerated files check passed"
  exit 0
else
  echo "Autogenerated files check failed. Please run \`configlet generate\` and commit in the changes."
  exit 1
fi  
```

Reply:
Thanks, that's better. I had to wrap `echo $GIT_DIFF` into `echo "$GIT_DIFF"`, otherwise new lines weren't printed.

Reply:
Oh that's a great trick. I think I should have used that in that gettext PR at work -_-

Reply:
Oooh, even fancier. I should really invest more time learning shell syntax

Reply:
Ah wait, but `git diff` always succeeds here, it doesn't fail when there is no difference.

Reply:
I guess that's Angelika added the `--exit-code` option.